pkgs/applications/editors/notepad-next/default.nix

```diff
@@ -0,0 +1,36 @@
+{ mkDerivation, lib, fetchFromGitHub, qmake, libsForQt5 }:
+
+mkDerivation rec {
+  pname = "notepad-next";
+  version = "0.5.1";
+
+  src = fetchFromGitHub {
+    owner = "dail8859";
+    repo = "NotepadNext";
+    rev = "v${version}";
+    sha256 = "sha256-J7Ngt6YtAAZsza2lN0d1lX3T8gNJHp60sCwwaLMGBHQ=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ qmake libsForQt5.qt5.qttools ];
+  qmakeFlags = [ "src/NotepadNext.pro" ];
+
+  # References
+  #  https://github.com/dail8859/NotepadNext/blob/master/doc/Building.md
+  #  https://github.com/dail8859/NotepadNext/issues/99
+  postPatch = ''
+    substituteInPlace ./src/NotepadNext/NotepadNext.pro --replace /usr $out
+  '';
+
+  CXXFLAGS = "-std=gnu++1z";
```

Comment:
Why is this necessary? Could not upstream set appropriate C++ requirements in their build scripts?

Reply:
I got this hint here: https://github.com/dail8859/NotepadNext/issues/99#issuecomment-1100762910
It decreases the amount of warnings while building...

Reply:
They should probably change https://github.com/dail8859/NotepadNext/blob/8bcfc93ba41f8bc1150385864f8adf501d842762/src/Config.pri#L24 to what C++ standards they expect, then.

Reply:
Thanks reported it upstream and keep track of it in the follow releases as package-maintainer :)